### PR TITLE
New var.storage_class

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -29,6 +30,7 @@ resource "google_project_service" "storage_api" {
 resource "google_storage_bucket" "gcs_bucket" {
   name                        = local.bucket_name
   location                    = local.bucket_location
+  storage_class               = local.bucket_storage_class
```
Comment (syedrakib) [inline main.tf]: this can be `var.storage_class ` itself. As we are not transforming it to anything else in local.

**Review by syedrakib — DISMISSED** [review]

File: variables.tf [hunk 2]
```diff
@@ -26,6 +26,12 @@ variable "location" {
   default     = ""
 }
 
+variable "storage_class" {
+  description = "The storage class of the GCS bucket. Defaults to 'Standard' if nothing is specified here. See https://cloud.google.com/storage/docs/storage-classes."
```
Comment (mfarix) [inline variables.tf]: All the available options are not mentioned in the doc. I guess you can consider including this line too: `Available options include "STANDARD", "MULTI_REGIONAL", "REGIONAL", "NEARLINE", "COLDLINE", "ARCHIVE"`

Comment (syedrakib) [reply variables.tf]: Regional Multi-Regional is not a storage "**CLASS**". Ref: https://cloud.google.com/storage/docs/storage-classes

File: variables.tf [hunk 3]
```diff
@@ -26,6 +26,12 @@ variable "location" {
   default     = ""
 }
 
+variable "storage_class" {
+  description = "The storage class of the GCS bucket. Defaults to 'STANDARD' if nothing is specified here. Available options include 'STANDARD', 'MULTI_REGIONAL', 'REGIONAL', 'NEARLINE', 'COLDLINE', 'ARCHIVE'. See https://cloud.google.com/storage/docs/storage-classes."
```
Comment (syedrakib) [inline variables.tf]:
```suggestion
  description = "The storage class of the GCS bucket. Defaults to 'STANDARD' if nothing is specified here. Available options include 'STANDARD', 'NEARLINE', 'COLDLINE', 'ARCHIVE'. See https://cloud.google.com/storage/docs/storage-classes."
```

Regional, Multi-Regional are not storage "CLASS". Ref: https://cloud.google.com/storage/docs/storage-classes

Comment (ghost) [reply variables.tf]: It's in additional classes https://cloud.google.com/storage/docs/storage-classes#legacy. Seems like it is some legacy class and we should be using STANDARD instead anyways. Will amend.

**Review by syedrakib — APPROVED** [review]

File: variables.tf (hunk 2 above)
Comment (mfarix) [reply variables.tf]: @syedrakib If you scroll down on the same doc, you will see a section for [Additional classes](https://cloud.google.com/storage/docs/storage-classes#legacy). But it seems like they are legacy and recommendation is to use STANDARD.

**Comment by mfarix** [comment]
LGTM

**Pull request merged**